File: ride.cpp
/*
ID: jgwilso1
PROG: ride
LANG: C++                 
*/

#include <iostream>
#include <string>
#include <map>

using namespace std;

int main() {
    freopen("ride.in", "r", stdin);
    freopen("ride.out", "w", stdout);

    string comet;
    string group;
    string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    int comet_id = 1;
    int group_id = 1;
    map<char,int> hashtable;

    getline(cin, comet);
    getline(cin, group);
    
    for (int i = 0; i < 26; i++) {
        hashtable[alphabet[i]] = (i + 1);
    }
    for (auto c : comet) {
        comet_id *= hashtable[c];
    }
    for (auto c : group) {
        group_id *= hashtable[c];
    }
    if (comet_id % 47 == group_id % 47) {
        cout << "GO\n";
    }
    else {
        cout << "STAY\n"; 
    }

    return 0;
}

File: usaco/training/ride.py
"""
PROG: ride
LANG: PYTHON3
"""
fin = open('ride.in', 'r')
fout = open('ride.out', 'w')
comet = fin.readline().strip('\n')
group = fin.readline().strip('\n')

chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
hashtable = {}
i = 0
for c in chars:
    i += 1
    hashtable[c] = i

comet_num = 1
group_num = 1
for c in comet:
    comet_num *= hashtable[c]
for c in group:
    group_num *= hashtable[c]

if group_num % 47 == comet_num % 47:
    verdict = 'GO'
else:
    verdict = 'STAY'
    
print(verdict)
fout.write(verdict + '\n')
fout.close()
